packages/merge-styles/src/Stylesheet.ts

```diff
@@ -154,9 +192,28 @@ export class Stylesheet {
    * Configures a reset callback.
    *
    * @param callback - A callback which will be called when the Stylesheet is reset.
+   * @returns function which when called un-registers provided callback.
    */
-  public onReset(callback: () => void): void {
+  public onReset(callback: Function): Function {
     this._onResetCallbacks.push(callback);
+
+    return () => {
+      this._onResetCallbacks = this._onResetCallbacks.filter(cb => cb !== callback);
+    };
+  }
+
+  /**
+   * Configures an insert rule callback.
+   *
+   * @param callback - A callback which will be called when a rule is inserted.
+   * @returns function which when called un-registers provided callback.
+   */
+  public onInsertRule(callback: Function): Function {
```

Comment:
This supersedes the likewise named listener in the config as it allows better support for multiple listeners to be registered separately, which is beneficial as we are a singleton and don't want random pieces of code to override our somebody's listener.
Returns an `unregister` function similarly to `onReset`